 <launch>

    <!-- Licensing Information: You are free to use or extend these projects for
      -  for education or research purposes provided that (1) you retain this notice
      -  and (2) you provide clear attribution to UC Berkeley, including a link 
      -  to http://barc-project.com
      -
      - Authors: J. Noonan and Jon Gonzales
      - Emails:  jpnoonan@berkeley.edu, jon.gonzales@berkeley.edu
    -->
    
    <!-- SYSTEM MODEL -->
    <!-- vehicle parameters -->
    <param name="m" type="double" value="1.98" />
    <param name="I_z" type="double" value="0.24" />
    <param name="L_a" type="double" value="0.125" />
    <param name="L_b" type="double" value="0.125" />
    <!-- control parameters -->
    <param name="input_gain" type="double" value="0.3" />

    <!-- ARDUINO -->
    <!--  * encoders and ultrasound sensors -->
    <node pkg="rosserial_python" type="serial_node.py" name="arduino_node" >
        <param name="port" value="/dev/ttyUSB0" />
    </node>

    <!-- LOW LEVEL CONTROLLER -->
    <node pkg="barc" type="controller_low_level.py" name="low_level_controller" output="screen"/>

	<!-- HIGH LEVEL CONTROLLER -->
	<node pkg="barc" type="servo_angle_controller_straight.py" name="controller" output="screen">
         <!-- PROPERTIES -->
        <param name="pwm_target" type="int" value="100" />
        <param name="yaw_target" type="int" value="90" />
        <param name="t_0" type="int" value="1" />
		<param name="t_f" type="int" value="9" />
		
        <!-- PID for straight test using imu gyro 0.022-->
        <param name="p" type="double" value="0.001" /> 
		<param name="i" type="double" value="-0.0002" />
		<param name="d" type="double" value="-0.001" />

		<param name="rate" type="int" value="50" />
   </node>

</launch>
